modules/core/m_mode.c

```diff
@@ -368,8 +376,32 @@ ms_bmask(struct MsgBuf *msgbuf_p, struct Client *client_p, struct Client *source
 						parv[3][0], s, forward);
 		}
 
-		if(add_id(fakesource_p, chptr, s, forward, banlist, mode_type))
+
+		if((banptr = add_id(fakesource_p, chptr, s, forward, banlist, mode_type)) != NULL)
 		{
+			if (extended)
+			{
+				bants = atol(strtok(NULL, " "));
+				who = strtok(NULL, " ");
+
+				if (who == NULL)
+				{
+					/* EBMASK params don't divide by 3, so we have an incomplete chunk */
+					sendto_realops_snomask(SNO_GENERAL, L_NETWIDE,
+						"Link %s dropped, invalid EBMASK chunk", source_p->name);
+					snprintf(squitreason, sizeof squitreason, "Invalid EBMASK chunk");
+					exit_client(source_p, source_p, source_p, squitreason);
+					return;
+				}
+
+				banptr->when = bants;
+				rb_free(banptr->who);
+				banptr->who = rb_strdup(who);
+
+				sprintf(dbuf, "%s ", s);
```

Comment:
```suggestion
				arglen = sprintf(dbuf, "%s ", s);
```

Reply:
Alternate suggestion (and then remove the next line, which I can't do in this suggestion): 

```suggestion
				dbuf += sprintf(dbuf, "%s ", s);
```